utils/s2n_random.c

```diff
@@ -337,24 +332,27 @@ int s2n_get_rdrand_data(struct s2n_blob *out)
 
     GUARD(s2n_stuffer_init(&stuffer, out));
     while ((space_remaining = s2n_stuffer_space_remaining(&stuffer))) {
-        int success = 0;
+        unsigned char success = 0;
         output.u64 = 0;
 
         for (int tries = 0; tries < 10; tries++) {
+            /* yeah I'm not buying the whole "we can't use the mnemonic name instead of the opt-code,
+             * we're already using a C99 compiler for Zeus sake. */
 #if defined(__i386__)
-            int success_high = 0, success_low = 0;
-            __asm__ __volatile__(".byte 0x48;\n" ".byte 0x0f;\n" ".byte 0xc7;\n" ".byte 0xf0;\n" "adcl $0x00, %%ebx;\n":"=b"(success_low), "=a"(output.i386_fields.u_low)
-                                 :"b"(0)
+            unsigned char success_high = 0, success_low = 0;
+
+            __asm__ __volatile__("rdrand %0;\n" "setc %1;\n": "=r"(output.i386_fields.u_low), "=qm"(success_low)
```

Comment:
I'm not familiar with inline assembly. Can you add a comment with a couple of sentences about what this assembly is doing?

Reply:
Specifically, I can guess what the keyword `rdrand` is doing, but I don't know what `setc`, `=r`, `=qm`, `r`, or `cc` are doing. Can you add a documentation link, or give a quick description about what this assembly is doing and how to read it?

Reply:
done